### PR TITLE
Fix ssl.certificates response

### PR DESCRIPTION
**Pull request by philkra**
as titled.

### CODE REVIEW

File: output/schema/schema.json [hunk 1]
```diff
@@ -51886,9 +51886,18 @@
...
+                "name": "ClusterCertificateInformation",
+                "namespace": "x_pack.ssl.get_certificates"
+              }
+            }
+          ],
           "type": {
-            "name": "ResponseBase",
-            "namespace": "common_abstractions.response"
+            "name": "Array",
+            "namespace": ".Users.pkrauss.github.com.elastic-client-generator.specification.node_modules.typescript.lib"
```
Comment (Mpdreamz) [inline output/schema/schema.json:51900]: What a freaky namespace?! cc @swallez @delvedor any clue how this one made it in ?

**Review by Mpdreamz — CHANGES_REQUESTED** [review]

File: output/schema/schema.json (hunk 1 above)
Comment (delvedor) [reply output/schema/schema.json:51900]: It's "correct", as `Array` is a native typescript type. I think @philkra wanted to use [`ArrayResponseBase`](https://github.com/elastic/elastic-client-generator/blob/master/docs/behaviors.md#arrayresponsebase).

File: output/typescript/types.ts [hunk 2]
```diff
@@ -4677,8 +4677,7 @@ export interface GetCategoriesResponse extends ResponseBase {
 export interface GetCertificatesRequest extends RequestBase {
 }
 
-export interface GetCertificatesResponse extends ResponseBase {
-  certificates: Array<ClusterCertificateInformation>
+export interface GetCertificatesResponse extends Array<ClusterCertificateInformation> {
```
Comment (delvedor) [inline output/typescript/types.ts:4680]:
```suggestion
export interface GetCertificatesResponse
  extends ResponseBase
  implements ArrayResponseBase<ClusterCertificateInformation> {
```

File: output/schema/schema.json (hunk 1 above)
Comment (philkra) [reply output/schema/schema.json:51900]: Fixed, @delvedor was on point

File: specification/specs/x_pack/ssl/get_certificates/GetCertificatesResponse.ts [hunk 3]
```diff
@@ -17,7 +17,5 @@
  * under the License.
  */
 
-@class_serializer('GetCertificatesResponseFormatter')
-class GetCertificatesResponse extends ResponseBase {
-  certificates: ClusterCertificateInformation[]
-}
+class GetCertificatesResponse
+  implements ArrayResponseBase<ClusterCertificateInformation> {}
```
Comment (delvedor) [inline specification/specs/x_pack/ssl/get_certificates/GetCertificatesResponse.ts:21]: I think this should extend from `ResponseBase` as well (cc @Mpdreamz)

Comment (Mpdreamz) [reply specification/specs/x_pack/ssl/get_certificates/GetCertificatesResponse.ts:21]: ++, it should. Another possible validation @swallez :)

Comment (swallez) [reply specification/specs/x_pack/ssl/get_certificates/GetCertificatesResponse.ts:21]: What is the effect of not extending `ResponseBase`? Is this marker interface used to identify endpoint responses?

Comment (Mpdreamz) [reply specification/specs/x_pack/ssl/get_certificates/GetCertificatesResponse.ts:21]: Yeah, we want to be in a position where each api invocation can be thought of as 

`client.api(RequestBase): ResponseBase`

That way clients have a path for abstraction to

`transport.request(....) ...`

**Pull request merged**